File: java/src/main/java/vsdk/toolkit/common/linealAlgebra/Vector2D.java
package vsdk.toolkit.common.linealAlgebra;

public class Vector2D {
    public double x;
    public double y;

    public Vector2D() {
        this(0.0, 0.0);
    }

    public Vector2D(double x, double y) {
        this.x = x;
        this.y = y;
    }

    @Override
    public String toString() {
        return "Vector2D{" +
            "x=" + x +
            ", y=" + y +
            '}';
    }
}

File: typescript/src/main/vsdk/toolkit/common/linealAlgebra/Vector2D.ts
export class Vector2D {
  public x: number;
  public y: number;

  public constructor(x = 0.0, y = 0.0) {
    this.x = x;
    this.y = y;
  }

  public static difference(a: Vector2D, b: Vector2D, o: Vector2D): void {
    o.x = a.x - b.x;
    o.y = a.y - b.y;
  }

  public static norm2(d: Vector2D): number {
    return d.x * d.x + d.y * d.y;
  }

  public toString(): string {
    return `Vector2D{x=${this.x}, y=${this.y}}`;
  }
}
